content/docs/arvo/clay/marks/examples.md

```diff
@@ -0,0 +1,248 @@
++++
+title = "Examples"
+weight = 4
+template = "doc.html"
++++
+
+These are the files used in the [Writing Marks](/docs/arvo/clay/marks/writing-marks) guide.
+
+## `/lib/csv/hoon`
+
+```hoon
+|%
+++  validate                                             ::  All rows same length?
```

Comment:
This is more of a side comment than a suggestion that you may have already considered, but it could potentially simplify some things to use `$|` to get a more restricted `(list (list @t))` type where all rows are the same length rather than calling `validate` in several locations. I haven't actually worked this through yet, and there might be pedagogical or stylistic reasons to use `(list (list @t))` rather than `$|  (list (list @t))  validate`, but perhaps it could be included as an exercise at the end if it actually makes sense to do so.

Reply:
oh true I wasn't familiar with `$|` tbh, I'll take a look and see if I can change it to using that